src/main/java/ru/touchin/roboswag/core/utils/diff/DiffUtil.java

```diff
@@ -0,0 +1,786 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ru.touchin.roboswag.core.utils.diff;
+
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * DiffUtil is a utility class that can calculate the difference between two lists and output a
+ * list of update operations that converts the first list into the second one.
+ * <p>
+ * It can be used to calculate updates for a RecyclerView Adapter.
+ * <p>
+ * DiffUtil uses Eugene W. Myers's difference algorithm to calculate the minimal number of updates
+ * to convert one list into another. Myers's algorithm does not handle items that are moved so
+ * DiffUtil runs a second pass on the result to detect items that were moved.
+ * <p>
+ * If the lists are large, this operation may take significant time so you are advised to run this
+ * on a background thread, get the {@link DiffResult} then apply it on the RecyclerView on the main
+ * thread.
+ * <p>
+ * This algorithm is optimized for space and uses O(N) space to find the minimal
+ * number of addition and removal operations between the two lists. It has O(N + D^2) expected time
+ * performance where D is the length of the edit script.
+ * <p>
+ * If move detection is enabled, it takes an additional O(N^2) time where N is the total number of
+ * added and removed items. If your lists are already sorted by the same constraint (e.g. a created
+ * timestamp for a list of posts), you can disable move detection to improve performance.
+ * <p>
+ * The actual runtime of the algorithm significantly depends on the number of changes in the list
+ * and the cost of your comparison methods. Below are some average run times for reference:
+ * (The test list is composed of random UUID Strings and the tests are run on Nexus 5X with M)
+ * <ul>
+ *     <li>100 items and 10 modifications: avg: 0.39 ms, median: 0.35 ms
+ *     <li>100 items and 100 modifications: 3.82 ms, median: 3.75 ms
+ *     <li>100 items and 100 modifications without moves: 2.09 ms, median: 2.06 ms
+ *     <li>1000 items and 50 modifications: avg: 4.67 ms, median: 4.59 ms
+ *     <li>1000 items and 50 modifications without moves: avg: 3.59 ms, median: 3.50 ms
+ *     <li>1000 items and 200 modifications: 27.07 ms, median: 26.92 ms
+ *     <li>1000 items and 200 modifications without moves: 13.54 ms, median: 13.36 ms
+ * </ul>
+ * <p>
+ * Due to implementation constraints, the max size of the list can be 2^26.
+ */
+public class DiffUtil {
+
+    private DiffUtil() {
```

Comment:
staticAnalysis would fail?
if you dont want to suppress it all then modify buildscripts to exclude everything with package like android.*